README.md

```diff
@@ -118,16 +108,48 @@ Fired when the scroller has finished rendering a new range of items, e.g. becaus
 
 Also see [the example below](#performing-actions-as-the-scroller-scrolls-using-the-rangechange-event).
 
+### The `ItemSource` class
+
+The `ItemSource` class represents a way of translating indices into JavaScript values. You can create them like so:
+
+```js
+const source = new ItemSource({
+  item(index) { ... },
+  getLength() { ... },
+  key(index) { ... }
+});
+```
+
+For example, to create an `ItemSource` that gets its items from a `contacts` array, and uses `contact.id` as the key, you could do
+
+```js
+const contactsSource = new ItemSource({
+  item(index) { return contacts[index]; },
+  getLength() { return contacts.length; },

```

Comment:
Then `get length() { ... }`

Reply:
That wouldn't work, because then we'd just invoke the getter once, in the constructor, and not have a reusable function we could call multiple times.